doomed_patient/doomed_patient/PatientGame.cpp

```diff
@@ -108,7 +116,24 @@ void PatientGame::handleEvents()
 // Game updates that need to happen every frame go here
 void PatientGame::update()
 {
+	for each (std::shared_ptr<Character> npc in level.characters)
+	{
+		// TODO: Bug where centre appears to be checking wrong cell for north and west edges
+		LevelCell::Passable direction = {
+			level.getCell((npc->getCentre()).convertWindowToGrid() )->getEdge(Directions::Direction::NORTH)->isWall(),
+			level.getCell(npc->getCentre().convertWindowToGrid())->getEdge(Directions::Direction::EAST)->isWall(),
+			level.getCell(npc->getCentre().convertWindowToGrid())->getEdge(Directions::Direction::SOUTH)->isWall(),
+			level.getCell((npc->getCentre()).convertWindowToGrid())->getEdge(Directions::Direction::WEST)->isWall()
+		};
```

Comment:
As it scans the cell edge when looking for a clear passage when it scans the North and West edge the npc skips the entire cell when approaching from the South or East. I think its due to the way that the edges are currently set-up in my build. And I think thats explaining it properly but don't quote me on that.